### PR TITLE
Add nightly fastlane firebase build

### PR DESCRIPTION
**Pull request by b-onc**
From now on, we should use `bundle exec fastlane match_me` to match our certificates and profiles!

**Please clone/build this branch locally to make sure code signing settings work!!!**

Nightly build will run at 2AM UTC everyday and will push a commit on master with message
> Build bump v<build num>

with increased build number

### CODE REVIEW

**Comment by VojtaStavik** [comment]
This is great work @b-onc! Just one detail - I don't think committing a build number back to the repo is necessary.

I'd suggest using a timestamp for the build number. Benefits:
  - It increments automatically and will always be unique
  - it identifies the exact time when the build was created
  - you don't need to pollute the repo with build number bump commits.

File: fastlane/Fastfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  
+  message = changelog_from_git_commits(commits_count: 10)
+  
+  firebase_app_distribution(
+      app: "1:674907137625:ios:10131789dab2e7cad7f348",
+      groups: "ios-stream-testers",
+      release_notes: message
+  )
+  
+  build_number = increment_build_number(xcodeproj: "./Example/Carthage/ChatExample.xcodeproj")
```
Comment (VojtaStavik) [inline fastlane/Fastfile:59]: I'd suggest doing this instead:
```
  # Set the build number to the current date (UTC, 24h) -> "20200419.0910"
  build_number = Time.new.utc.strftime("%Y%m%d.%H%M")
```

File: fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  
+  commit_version_bump(
```
Comment (VojtaStavik) [inline fastlane/Fastfile:61]: I think we should remove this.

File: fastlane/Fastfile (hunk 1 above)
Comment (VojtaStavik) [reply fastlane/Fastfile:59]:
```
  increment_build_number(
    xcodeproj: "./Example/Carthage/ChatExample.xcodeproj",
    build_number: build_number
  )
```

**Comment by VojtaStavik** [comment]
Another idea - do you think it's necessary to create a new build every day? What if there was no change to the repo since? Why should we create a new build which is exactly the same as the previous one?

Maybe we should rather create a new test build for every merge to master. We will have multiple builds a day for busy days but no builds when there's nothing happening in the repo. I think it makes better sense like this and it matches nicely the development activity.

**Comment by buh** [comment]
I agree with @VojtaStavik the build should be on merge to master 💯

**Pull request merged**